api/tests/routes/backoffice_v3/venues_test.py

```diff
@@ -901,3 +909,100 @@ def test_venue_without_history(self, authenticated_client, legit_user):
+            "select", attrs={"data-tomselect-autocomplete-url": "/autocomplete/criteria"}
+        )
+        assert json.loads(autocomplete_select.attrs["data-tomselect-options"]) == [
+            {"id": str(criteria[1].id), "text": criteria[1].name}
+        ]
+        assert json.loads(autocomplete_select.attrs["data-tomselect-items"]) == [str(criteria[1].id)]
+
+
+class BatchEditVenuesTest(PostEndpointHelper):
+    endpoint = "backoffice_v3_web.venue.batch_edit_venues"
+    endpoint_kwargs = {"object_ids": "1,2"}
+    needed_permission = perm_models.Permissions.MANAGE_PRO_ENTITY
+
+    def test_empty_batch_edit_venues(self, legit_user, authenticated_client):
+        form_data = {
+            "object_ids": "",
+            "all_permanent": "",
+            "all_not_permanent": "",
+        }
+
+        response = self.post_to_endpoint(authenticated_client, form=form_data)
+        assert response.status_code == 303
+
+        redirected_response = authenticated_client.get(response.location)
+        assert "L'un des identifiants sélectionnés est invalide" in html_parser.extract_alert(redirected_response.data)
+
+    @pytest.mark.parametrize("is_permanent", [True, False])
+    def test_batch_edit_venues_two_checkboxes(self, legit_user, authenticated_client, is_permanent):
+        venue = offerers_factories.VenueFactory(isPermanent=is_permanent)
+
+        form_data = {
+            "object_ids": str(venue.id),
+            "criteria": "",
+            "all_permanent": "on",
+            "all_not_permanent": "on",
+        }
+
+        response = self.post_to_endpoint(authenticated_client, form=form_data)
+        assert response.status_code == 303
+
+        redirected_response = authenticated_client.get(response.location)
+        assert "Impossible de passer tous les lieux en permanents et non permanents" in html_parser.extract_alert(
+            redirected_response.data
+        )
+
+        assert venue.isPermanent is is_permanent  # unchanged
+
+    @pytest.mark.parametrize("set_permanent", [True, False])
+    def test_batch_edit_venues(self, legit_user, authenticated_client, criteria, set_permanent):
+        new_criterion = criteria_factories.CriterionFactory()
+        venues = [
+            offerers_factories.VenueFactory(criteria=criteria[:2], isPermanent=set_permanent),
+            offerers_factories.VenueFactory(criteria=criteria[1:], isPermanent=not set_permanent),
+        ]
+
+        form_data = {
+            "object_ids": ",".join(str(venue.id) for venue in venues),
+            "criteria": [criteria[0].id, new_criterion.id],
+            "all_permanent": "on" if set_permanent else "",
+            "all_not_permanent": "" if set_permanent else "on",
+        }
+
+        response = self.post_to_endpoint(authenticated_client, form=form_data)
+        assert response.status_code == 303
+
+        assert set(venues[0].criteria) == {criteria[0], new_criterion}  # 1 kept, 1 removed, 1added
```

Comment:
```suggestion
        assert set(venues[0].criteria) == {criteria[0], new_criterion}  # 1 kept, 1 removed, 1 added
```
Victoire, j'ai un commentaire !